app/src/main/java/org/mozilla/fenix/settings/search/AddSearchEngineFragment.kt

```diff
@@ -170,7 +170,7 @@ class AddSearchEngineFragment : Fragment(), CompoundButton.OnCheckedChangeListen
         if (searchString.isEmpty()) {
             custom_search_engine_search_string_field
                 .error = resources.getString(R.string.search_add_custom_engine_error_empty_search_string)
-            hasError = true
+            return
```

Comment:
This definitely fixes the overwriting problem, but I feel like the early return also adds to the complexity of the code.  I would prefer if we could link these two error checks (`isEmpty` and `contains`) to show that we only expect to handle one at a time.  For example, maybe something along these lines.

```kotlin
val searchFieldError = when {
     searchString.isEmpty() -> R.string.search_add_custom_engine_error_empty_search_string
    !searchString.contains("%s") -> R.string.search_add_custom_engine_error_missing_template
    else -> null
}
if (searchFieldError != null) {
    custom_search_engine_search_string_field.error = resources.getString(searchFieldError)
    hasError = true
}
```

Reply:
@Baron-Severin This fixes the visual problem. The cutoff is due to a known bug in Android that cuts the text when error is set multiple times, like in this case. The fix in this PR is intended to fix both the visual bug and the logic of error handling for the field.

Reply:
Huh, TIL.  Thanks @mcarare!  Approving with this readability change as an optional nit, then.